src/main/java/me/magnet/consultant/ConfigUpdater.java

```diff
@@ -112,16 +122,25 @@ public void run() {
 			}
 		}
 		catch (IOException | RuntimeException e) {
+			if (interruptedBecauseShutdown(e)) {
+				return;
+			}
 			log.error("Error occurred while retrieving/publishing new config from Consul: " + e.getMessage(), e);
 		}
 		finally {
-			ConfigUpdater task = new ConfigUpdater(executor, httpClient, consulURI, consulIndex, identifier,
-					objectMapper, config, listener, kvPrefix);
-
-			executor.schedule(task, timeout, TimeUnit.MILLISECONDS);
+			if (!shutdownBegun.get()) {
+				log.info("Marked for shutdown. Not running");
+				executor.schedule(this, timeout, TimeUnit.MILLISECONDS);
+			}
 		}
 	}
 
+	private boolean interruptedBecauseShutdown(Exception e) {
+		return e instanceof IllegalStateException
+				&& "Connection pool shut down".equals(e.getMessage())
```

Comment:
If it's the get that's aborted, the exception will be `InterruptedIOException`. Might be worth testing for that one?